server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1136,7 +1137,15 @@ private static void validateCompositeTemplate(final ClusterState state,
                         validateTimestampFieldMapping(tsFieldName, mapperService);
                     }
                 } catch (Exception e) {
-                    throw new IllegalArgumentException("invalid composite mappings for [" + templateName + "]", e);
+                    final String expectedMessageIfDataStreamsIsUsedAndNotSupported =
```

Comment:
Instead of relying on an exception with a specific message, what if we proactively checked if the timestamp mapper existed? We could do something like

```
boolean timestampMapperExists = tempIndexService.mapperService().isMetadataField("_timestamp");
```

and then throw if there's a `data_stream` definition but no `_timestamp` metadata mapper.

Reply:
Thanks. This is less fragile than checking the exception message.